### ISSUE TITLE
Crashing Visual Studio

### ISSUE THREAD

**Issue by janosNH** [madskristensen/MarkdownEditor#64]
### Installed product versions
Microsoft Visual Studio Community 2015
Version 14.0.25422.01 Update 3
Microsoft .NET Framework
Version 4.6.01055
Installed Version: Community
Extensibility Tools   1.10.184
Markdown Editor   1.8.166

### Description
Typing a + sign at the beginning of a line crashes the Markdown Editor and VS. 

This is the exception:
Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
```
Exception Info: System.NullReferenceException
   at Markdig.Extensions.Tables.GridTableParser.TryOpen(Markdig.Parsers.BlockProcessor)
   at Markdig.Parsers.BlockProcessor.TryOpenBlocks(Markdig.Parsers.BlockParser[])
   at Markdig.Parsers.BlockProcessor.TryOpenBlocks()
   at Markdig.Parsers.BlockProcessor.ProcessLine(Markdig.Helpers.StringSlice)
   at Markdig.Parsers.MarkdownParser.ProcessBlocks()
   at Markdig.Parsers.MarkdownParser.Parse()
   at Markdig.Parsers.MarkdownParser.Parse(System.String, Markdig.MarkdownPipeline)
   at Markdig.Markdown.Parse(System.String, Markdig.MarkdownPipeline)
   at MarkdownEditor.Parsing.MarkdownFactory+<>c__DisplayClass6_0.<ParseToMarkdown>b__0(Microsoft.VisualStudio.Text.ITextSnapshot)
   at System.Runtime.CompilerServices.ConditionalWeakTable`2[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].GetValue(System.__Canon, CreateValueCallback<System.__Canon,System.__Canon>)
   at MarkdownEditor.Parsing.MarkdownFactory.ParseToMarkdown(Microsoft.VisualStudio.Text.ITextSnapshot, System.String)
   at MarkdownEditor.MarkdownClassifier.<ParseDocument>b__17_0()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
   at MarkdownEditor.MarkdownClassifier+<ParseDocument>d__17.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr)
```

**Comment by xoofx** [comment]
Thanks for the report. Out of curiosity, how did you get this exception log? (did you go to ActivityLog.xml?)

**Comment by janosNH** [comment]
I got it from Event Viewer\Windows Logs\Application.

 

Janos

**Comment by xoofx** [comment]
This should be fixed by commit 4104b99. We will try to improve the exception report experience and avoid the crashing through the issue #63
